### ISSUE TITLE
DataLoader加载MNIST数据集后，label值出现错误，什么原因？

### ISSUE THREAD

**Issue by zh82510** [PaddlePaddle/Paddle#67531]
### 请提出你的问题 Please ask your question

PaddlePaddle版本：2.6
Python版本： 3.9.0
系统：Windows 11

`import paddle
import paddle.vision.transforms as T
from paddle.vision.datasets  import MNIST
from paddle.io import DataLoader

paddle.device.set_device("gpu:0")
mnist = MNIST(mode="train", transform=T.ToTensor())
for i in range(5):
    img, label = mnist[i]
    print(img.shape, label)

train_loader = DataLoader(mnist, batch_size=5, shuffle=False)
for idx,data in enumerate(train_loader()):
    img= data[0]
    label= data[1]
    if idx == 0:
        print("DataLoader img.shape:{},\nDataLoader label:{}".format(img.shape,label))
        break`

CUDA版本是11.6；cuDNN版本是8.4.0。

程序运行的结果：
![屏幕截图 2024-08-19 103317](https://github.com/user-attachments/assets/3c020716-8733-46df-810c-d5f3e30d027c)

问题：DataLoader加载MNIST数据集后，label的值应当也是[[5],[0],[4],[1],[9]]，Dataloader加载时设置的是不打乱顺序。但运行结果label值很奇怪，好像是随机的值。试了CUDA 11.7 + cuDNN 8.4.1，结果是同样的问题。请问是什么原因？谢谢。

**Comment by deepllz** [comment]
hi，你好，麻烦升级Paddle到最新版本，看下windows这个问题还是否存在。

**Comment by deepllz** [comment]
我用CUDA 11.7 版本的Paddle没有复现这个问题，可以重新安装一下paddle试试看。

**Comment by zh82510** [comment]
谢谢您的回复，按照您说的方法，在conda虚拟环境中卸载paddle2.6，安装paddle3.0 beta版，CUDA选的版本是11.8。
     python -m pip install paddlepaddle-gpu==3.0.0b1 -i https://www.paddlepaddle.org.cn/packages/stable/cu118/
之前系统下安装的CUDA11.6没动。测试结果显示label值正常。

之前使用的测试环境是paddle 2.6 + CUDA11.6。
步骤：
1.  在Windows系统下安装CUDA11.6；
2.  在conda的虚拟环境下安装paddle 2.6，选择paddle时选择CUDA11.6。

测试结果显示label值不正常，如上面问题中的截图。
新建一个虚拟环境，安装paddle2.6，CUDA也选11.8。测试结果和之前一样，label值不正常。

从上面的结果看是paddle2.6版本的原因？还是环境配置的问题？

**Comment by deepllz** [comment]
> 谢谢您的回复，按照您说的方法，在conda虚拟环境中卸载paddle2.6，安装paddle3.0 beta版，CUDA选的版本是11.8。 python -m pip install paddlepaddle-gpu==3.0.0b1 -i https://www.paddlepaddle.org.cn/packages/stable/cu118/ 之前系统下安装的CUDA11.6没动。测试结果显示label值正常。
> 
> 之前使用的测试环境是paddle 2.6 + CUDA11.6。 步骤：
> 
> 1. 在Windows系统下安装CUDA11.6；
> 2. 在conda的虚拟环境下安装paddle 2.6，选择paddle时选择CUDA11.6。
> 
> 测试结果显示label值不正常，如上面问题中的截图。 新建一个虚拟环境，安装paddle2.6，CUDA也选11.8。测试结果和之前一样，label值不正常。
> 
> 从上面的结果看是paddle2.6版本的原因？还是环境配置的问题？

从您的测试情况看，大概率是window版本的paddle存在某些问题，然后在最新的代码修复了。可以用paddle2.6 + cuda11.8验证一下是不是cuda环境的问题。
